### ISSUE TITLE
Twig Sandbox SecurityPolicy not working as intended?

### ISSUE THREAD

**Issue by Exhibitioner** [twigphp/Twig#2758]
I've been fiddling around with Twig's sandbox and the SecurityPolicy extension that compliments it, however cannot get the policies to work as they are intended to.

I've looked at online docs, and other examples but most uses of the SecurityPolicy make use of the Symphony framework's pseudo-code and I'm just using Twig as a standalone package.

For some odd reason, I can't allow properties by doing `'Class' => array('id', 'title')`, oddly if I substitute `"StdClass"` for the class name instead - no security exception is raised and the template renders fine; even though `get_class()` on the used variable returns the actual class name.

See this issue for more details: https://stackoverflow.com/questions/52569753/twig-sandbox-security-policy-wont-work

**Comment by Exhibitioner** [comment]
The exception I get when trying to use the correct class name in the properties argument is the following: `Twig_Sandbox_SecurityNotAllowedPropertyError: Calling "id" property on a "stdClass" object is not allowed.`

**Comment by xabbuh** [comment]
How exactly did you configure the sandbox when you got this exception?

**Comment by stof** [comment]
Well, your template does not operate on a `GetThings` object in your SO question, but on the return value of `doStuff`

**Comment by Exhibitioner** [comment]
Yep that is what I have found so far. It's an issue with how \PDO's \PDO::FETCH_OBJ works.

Unlike fetch associated array, it creates a StdClass and loads the returned data into that StdClass, before returning it. Now I'm wondering what's the best way to fix this.

Ultimately so that I can do 'GetThings' => array(), rather than the current convoluted workaround I'm using: 'StdClass' => array(). I'm open to suggestions. 

I tried \PDO::FETCH_CLASS, but there seems to be more issues here. First, the template renders fine without any restrictions i.e. I can have an empty allowedProperties and it'll still accept it. 

And fetch_class  returns double results, both associated and indexed for every column fetched ([id] => 2 [0] => 2 [data] => Abc [1] => Abc) which seems unnecessary and a waste.
